packages/broker/test/integration/plugins/operator/MaintainTopologyHelper.test.ts

```diff
@@ -20,166 +32,144 @@ describe('MaintainTopologyHelper', () => {
+        let nodeWallet: Wallet
         let sponsorship1: Contract
         let sponsorship2: Contract
         let topologyHelper: MaintainTopologyHelper
 
         beforeAll(async () => {
-            const deployment = await setupOperatorContract({ nodeCount: 1 })
-            operatorWallet = deployment.operatorWallet
-            operatorContract = deployment.operatorContract
-            operatorServiceConfig = {
-                ...deployment.operatorServiceConfig,
-                nodeWallet: deployment.nodeWallets[0]
-            }
+            ({
+                operatorWallet,
+                operatorContract,
+                operatorServiceConfig,
+                nodeWallets: [nodeWallet]
+            } = await setupOperatorContract({ nodeCount: 1 }))
+        }, TIMEOUT)
+
+        beforeEach(() => {
+            topologyHelper = new MaintainTopologyHelper({
+                ...operatorServiceConfig,
+                nodeWallet
+            })
         })
 
         afterEach(async () => {
             topologyHelper.stop()
-            operatorContract.provider.removeAllListeners()
         })
 
-        it('edge cases, 2 sponsorships for the same stream, join only fired once', async () => {
-
-            topologyHelper = new MaintainTopologyHelper(operatorServiceConfig)
-            let receivedAddStreams = 0
-            topologyHelper.on('addStakedStreams', () => {
-                receivedAddStreams += 1
-            })
-            await wait(2000)
+        it('emits event once when both sponsorships are staked to', async () => {
+            const events = eventsToArray(topologyHelper as any, ['addStakedStreams'])
             await topologyHelper.start()
 
-            sponsorship1 = await deploySponsorshipContract({ streamId: streamId1, deployer: operatorWallet })
-            sponsorship2 = await deploySponsorshipContract({ streamId: streamId1, deployer: operatorWallet })
-
-            await delegate(operatorWallet, operatorContract.address, 200)
+            await ensureMinElapsedTime(async () => {
+                sponsorship1 = await deploySponsorshipContract({ streamId: streamId1, deployer: operatorWallet })
+                sponsorship2 = await deploySponsorshipContract({ streamId: streamId1, deployer: operatorWallet })
+                await delegate(operatorWallet, operatorContract.address, 200)
+            }, WAIT_FOR_EVENT_HANDLERS_TO_REGISTER_IN_MS)
 
             await stake(operatorContract, sponsorship1.address, 100)
-            await waitForCondition(() => receivedAddStreams === 1, 10000, 1000)
-            await stake(operatorContract, sponsorship2.address, 100)
-            await waitForCondition(() => receivedAddStreams === 1, 10000, 1000)
-
-            await wait(10000) // wait for events to be processed
+            await waitForCondition(() => events.length >= 1, WAIT_FOR_EVENTS_IN_MS)
+            expect(events.length).toEqual(1)
```

Comment:
Should we check that the event contains info about the both streams? (also lines `158` and `172`)

Reply:
In this case it is actually just one stream since both sponsorships point to the same string.

If you're okay with it, I'll leave the value check out for now. We should most likely anyway turn most of these cases into unit tests in the future and keep the main happy path as an integration test for example.